tempel.el

```diff
@@ -221,6 +221,18 @@ BEG and END are the boundaries of the modification."
           (tempel--synchronize-fields st ov)))))
     (tempel--update-mark ov)))
 
+(defun tempel--prop-get (prop &optional element)
+  "Get the property PROP of element ELEMENT or the field at point."
+  (when-let ((element (or element (tempel--field-at-point)))
+             (prop (intern (format "tempel--prop-%s" prop))))
+    (overlay-get element prop)))
+
+(defun tempel--prop-put (prop value &optional element)
+  "Set the property PROP of ELEMENT or the element at point to VALUE."
+  (when-let ((element (or element (tempel--field-at-point)))
+             (prop (intern (format "tempel--prop-%s" prop))))
+    (overlay-put element prop value)))
```

Comment:
I would prefer if we remove these helper functions. It seems in the code below you always have a field given (the element, as you call it here, is non-nil). Then the calls reduce to just overlay-get/overlay-put with a symbol, which is more efficient and consistent with the rest of the code base. I attach multiple tempel--* properties to the overlays.